front/pages/w/[wId]/ds/index.tsx

```diff
@@ -95,32 +100,33 @@ export const getServerSideProps: GetServerSideProps<{
   const readOnly = !auth.isBuilder();
 
   let allDataSources = await getDataSources(auth);
-  const dataSources = allDataSources.filter((ds) => !ds.connector);
-  const managedDataSources = allDataSources.filter((ds) => ds.connector);
-
-  let connectorStatuses = managedDataSources.map(() => "failed");
-  try {
-    connectorStatuses = (
-      await Promise.all(
-        managedDataSources.map(({ connector }) =>
-          ConnectorsAPI.getSyncStatus(connector!.id)
-        )
-      )
-    ).map((s): ConnectorSyncStatus => (s.isErr() ? "failed" : "succeeded"));
-  } catch (e) {
-    // Probably means `connectors` is down, we log but don't fail to avoid a 500 when just
-    // displaying the datasources (eventual actions will fail but a 500 just at display is not
-    // desirable). When that happens the managed data sources are shown as failed.
-    logger.error({ error: e }, "Failed to retrieve connector status");
-  }
-
-  const connectorStatusByProvider: Record<
-    ConnectorProvider,
-    ConnectorSyncStatus
-  > = managedDataSources.reduce(
-    (acc, ds, i) =>
-      Object.assign(acc, { [ds.connector!.provider]: connectorStatuses[i] }),
-    {} as Record<ConnectorProvider, ConnectorSyncStatus>
+  const dataSources = allDataSources.filter((ds) => !ds.connectorId);
+  const managedDataSources = allDataSources.filter((ds) => ds.connectorId);
+  const provider2Connector = await Promise.all(
+    managedDataSources.map(async (mds) => {
+      try {
+        const statusRes = await ConnectorsAPI.getConnector(mds.connectorId!);
+        if (statusRes.isErr()) {
+          return {
+            provider: mds.connectorProvider,
+            connector: undefined,
+            fetchConnectorError: true,
+          };
+        }
+        return {
+          provider: mds.connectorProvider,
+          connector: statusRes.value,
+          fetchConnectorError: false,
+        };
+      } catch (e) {
```

Comment:
please preserve comment :pray:

Reply:
my bad

Reply:
Marked as resolved but still not there (post commit indicating fix)

Reply:
thank you!!!